src/goodHealthApp/models.py

```diff
@@ -1,16 +1,40 @@
 from django.contrib.auth.models import User
 from django.db import models
+from django.conf import settings
 from django.utils.timezone import now
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+def get_upload_image_dir(instance, filename):
+    return "uploads/{mimeType}/{userid}/{file}".format(mimeType="images", userid=instance.user.username, file=filename)
 
 
 class UserProfile(models.Model):
-    """
-    UserProfile Model.
-    """
     user = models.OneToOneField(User, on_delete=models.CASCADE)
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    email = models.CharField(max_length=100)
+    password = models.CharField(max_length=100)
+    phone_number = models.CharField(max_length=100)
+    avatar = models.ImageField(upload_to=get_upload_image_dir, null=True, blank=True)
+
+    def __str__(self):  # __unicode__ for Python 2
+        return f"{self.first_name} {self.last_name}"
```

Comment:
This should be changed too with respect to the required changes in the userprofile model